core/connection/manager.go

```diff
@@ -384,17 +384,17 @@ func (manager *connectionManager) createP2PSession(c Connection, p2pChannel p2p.
 		return session.SessionDto{}, session.PaymentInfo{}, fmt.Errorf("could not marshal session config: %w", err)
 	}
 
-	sessionRequest := p2p.ProtoMessage(&pb.SessionRequest{
+	sessionRequest := &pb.SessionRequest{
 		Consumer: &pb.ConsumerInfo{
 			Id:             consumerID.Address,
 			AccountantID:   accountantID.Address,
 			PaymentVersion: string(session.PaymentVersionV3),
 		},
 		ProposalID: int64(proposal.ID),
 		Config:     config,
-	})
-
-	res, err := p2pChannel.Send(p2p.TopicSessionCreate, sessionRequest)
+	}
+	log.Info().Msgf("Sending P2P message to %q: %s", p2p.TopicSessionCreate, sessionRequest.String())
+	res, err := p2pChannel.Send(p2p.TopicSessionCreate, p2p.ProtoMessage(sessionRequest))
```

Comment:
Please do logging inside `p2pChannel.Send()`, will be hard to remember to add this liners before each sending or receiving.
Also code will be less noisy.

Reply:
Send message accepts this struct
```go
type Message struct {
	Data []byte
}
```
It is not coupled to concrete message struct, you just send bytes. If I add it to Send I can't log formatted payload with fields. There are only 5 places there it is needed right now and we will not have much more of these messages.

Also, what if you want to add more context to logging?  I suggest to keep it explicit as is.

Reply:
Well serialisation should happen in transport level also :)

Reply:
p2p transport layer works with raw bytes, there is no need to for extra abstractions.